### PR TITLE
Add icon themes and cursor themes

### PR DESCRIPTION
**Pull request by 01micko**
This is my take on adding features to `'tweaks`

Notes:
- hard coded adwaita for cursors and gtk as it's builtin to gtk. Not much can be done about that
- icon themes and everything else works pretty well
- cursor theme isn't changing reliably so this is WIP

tweaks

If you get time @johanmalm or @consolatis you could rewrite this correctly - I'm pretty much a novice in `C`, certainly know enough to be dangerous!

### CODE REVIEW

**Comment by johanmalm** [comment]
Just had a click look.

`find_themes()` is getting quite hard to follow. It might be better to create a separate `find_cursor_themes()` to deal with associated edge-cases there.

Would we not to set `XCURSOR_THEME` in `~/.config/labwc/environment` for it to have an effect outside of GTK applications?

Some comments in the code regarding the hard-coded Adwaita would be helpful. I still get a `/usr/share/themes/Adwaita/` on my box. What's the case you're trying to deal with by hard-coding it? Are we wanting add to Adwaita even if an theme directory does not exist? If so, it might be easier to keep `find_themes()` clean (i.e. no Adwaita check). Then after it's been called, check if the vector contains Adwaita - and if not, then explicitly add it.

**Comment by 01micko** [comment]
> Just had a click look.
> 
> `find_themes()` is getting quite hard to follow. It might be better to create a separate `find_cursor_themes()` to deal with associated edge-cases there.
> 

Ok, agreed.

> Would we not to set `XCURSOR_THEME` in `~/.config/labwc/environment` for it to have an effect outside of GTK applications?
> 

Ok, I'll have to walk through that and see if it's set and either edit it or append it

> Some comments in the code regarding the hard-coded Adwaita would be helpful. I still get a `/usr/share/themes/Adwaita/` on my box. What's the case you're trying to deal with by hard-coding it?

As far as I understood it was the default _builtin_ gtk theme and it didn't show up in my tests so I hard coded it in. But now I suppose some distros separate it out (which makes sense).

> Are we wanting add to Adwaita even if an theme directory does not exist?

Yes

>  If so, it might be easier to keep `find_themes()` clean (i.e. no Adwaita check). Then after it's been called, check if the vector contains Adwaita - and if not, then explicitly add it.

Ok :)

I'll find some time next week end to clean this up. For now I'll leave it as a 'draft' and am willing to accept anyone's critique.

Thanks for looking.

**Comment by 01micko** [comment]
> Just had a click look.
> 
> `find_themes()` is getting quite hard to follow. It might be better to create a separate `find_cursor_themes()` to deal with associated edge-cases there.


Declined that, but `find_themes()` is simplified from my last effort.

> Would we not to set `XCURSOR_THEME` in `~/.config/labwc/environment` for it to have an effect outside of GTK applications?

Yes, and did so, but fails to be reloaded on my system. Need confirmation.

> Some comments in the code regarding the hard-coded Adwaita would be helpful.

Done.

>  I still get a `/usr/share/themes/Adwaita/` on my box. What's the case you're trying to deal with by hard-coding it? Are we wanting add to Adwaita even if an theme directory does not exist? If so, it might be easier to keep `find_themes()` clean (i.e. no Adwaita check). Then after it's been called, check if the vector contains Adwaita - and if not, then explicitly add it.

I don't have the GTK theme so hard coding is the only option (I could see), but I softened it accordingly. Code comments amended accordingly.

---

I also took the liberty to adjust `README` reflecting the changes and add other comments to the code.

**Comment by 01micko** [comment]
Converted to draft until I find time to fix conflicts.

**Comment by johanmalm** [comment]
Whoops. Forgot about this PR. Will have a look now to see if I can merge.

**Comment by 01micko** [comment]
Nah, I'll fix it and force push when ready. :-)

**Comment by johanmalm** [comment]
Whoops... read your note too late. Hopefully that's okay. Did a quick fix on a segfault.

**Comment by 01micko** [comment]
All good, thanks for the help :smiley:

**Comment by johanmalm** [comment]
Felt guilty :smile: 
Had completely forgotten about that PR. Short term memory is not what it used to be

**Comment by 01micko** [comment]
This is open source. We _help_ each other, that's what it's about, and you've been a big help to me and I thank you for that. I hope I've been somewhat of some help to you :smiley_cat:

**Comment by johanmalm** [comment]
Absolutely.
Love the cat-smiley to go with lab+puppy
